src/conf_mode/vpn_openconnect.py

```diff
@@ -63,17 +70,35 @@ def get_config():
 def verify(ocserv):
     if ocserv is None:
         return None
-
     # Check authentication
     if "authentication" in ocserv:
         if "mode" in ocserv["authentication"]:
             if "local" in ocserv["authentication"]["mode"]:
-                if not ocserv["authentication"]["local_users"] or not ocserv["authentication"]["local_users"]["username"]:
+                if "radius" in ocserv["authentication"]["mode"]:
+                    raise ConfigError('openconnect supports only one authentication mode. Currently configured \'local\' and \'radius\' modes')
+                if not ocserv["authentication"]["local_users"]:
+                    raise ConfigError('openconnect mode local required at leat one user')
+                if not ocserv["authentication"]["local_users"]["username"]:
                     raise ConfigError('openconnect mode local required at leat one user')
                 else:
-                    for user in ocserv["authentication"]["local_users"]["username"]:
-                        if not "password" in ocserv["authentication"]["local_users"]["username"][user]:
-                            raise ConfigError(f'password required for user {user}')
+                    # For OTP mode: verify that each local user has an OTP key
+                    if "otp" in ocserv["authentication"]["mode"]["local"]:
+                        users_wo_key = []
+                        for user in ocserv["authentication"]["local_users"]["username"]:
```

Comment:
You can have this easier and more readble using:


```Python
for user, user_config in ocserv["authentication"]["local_users"]["username"].items():
    # User has no OTP key defined
    if dict_search('otp.key', user_config) == None:
        users_wo_key.append(user)
```

Reply:
Fixed